internal/server/data/access_key.go

```diff
@@ -194,6 +196,9 @@ func GetAccessKey(tx ReadTxn, opts GetAccessKeysOptions) (*models.AccessKey, err
 	if opts.ByID > 0 {
 		query.B("and id = ?", opts.ByID)
 	}
+	if opts.ByName != "" {
+		query.B("and name = ?", opts.ByName)
```

Comment:
Ah, I missed this in the previous PR, but we are missing a query clause when we use `ByID` and `ByName`.

If `ByKeyID` is set we need to omit `organization_id = ?` because we need to query for all keys, we don't know the org yet.

Now that we've added `ByID` and `byName` though, we have to add `organization_id = ?` if either of those are set.

That might be a good reason to have two separate functions (one for `byKeyID`, and the other that takes options but always set an `organization_id`).  I'd also be fine keeping one function and adding 

```go
if opts.ByID != 0 || opts.ByName != "" {
    query.B("organization_id = ? , tx.OrganizationID())
}
```